### ISSUE TITLE
Sorting: Put numbers on top.

### ISSUE THREAD

**Issue by ponychicken** [owncloud/ios#561]
Both in the Mac Finder as well in Windows Explorer, files are by default sorted in a way putting numerals on top. The fact that the ownCloud iOS app doesn't stick to this convention makes it difficult to create custom sort orders for folders by prefixing them with 01, 02 etc.

Here are some screenshots of Explorer and Finder: 

 ![explorer on win 7](http://www.gieseke-buch.de/wordpress/wp-content/uploads/2009/09/w7F3_038.jpg)
![mac finder](https://cloud.githubusercontent.com/assets/183302/12449375/1e06cf6a-bf7d-11e5-9cef-f8b037d508aa.png)

**Comment by nasli** [comment]
@ponychicken thanks for your suggestion. 
Currently we have alphabetical index when there are more than 20 files and numbers and special characters are sorted in the last position (#), we could change it to be the first one

**Comment by ponychicken** [comment]
Yes, I think that would be in line with what users are used to. If you point my to where the logic sits I can submit a PR.


Am 11. Februar 2016 bei 10:15:30, Noelia Alvarez (notifications@github.com) schrieb:

@ponychicken thanks for your suggestion.
Currently we have alphabetical index when there are more than 20 files and numbers and special characters are sorted in the last position (#), we could change it to be the first one

**Comment by javiergonzper** [comment]
Hello @ponychicken 
On iOS the numbers are on the bottom. You can check it for example on the "Contacts".

![iphone-contacts-app-ios-7](https://cloud.githubusercontent.com/assets/1815050/12973593/c116d2d6-d0ad-11e5-9c67-473b01fee534.png)


In my opinion it is not a good a idea change the natural behaviour to a different one.

**Comment by rperezb** [comment]
Thanks for your suggestion, ftm, we plan to implement in the short term: #391 offer to sort data not only alphabetically but also by modification time as it's already implemented on Android app.
 
Let's keep this feature request in the backlog

**Comment by ponychicken** [comment]
@javiergonzper The example you choose is not a good one as apple employs a different sorting algorithm for persons names, as they are unlikely to have numbers in them. If you look at a file sorting, for example in the iCloud drive, files are sorted as i reported, numbers first. Attached a screen.
![image](https://cloud.githubusercontent.com/assets/183302/13033137/a205c55e-d30b-11e5-9e59-057342756808.png)

**Comment by javiergonzper** [comment]
@ponychicken but your example it is not a list. It is a grid.
Also the folders are at the beginning. If we set the files and folders in the order that you want we lose completely the right bar with the direct access to initial letter of the files.

**Comment by ponychicken** [comment]
Sorry if that was confusing, i'm not proposing to change the. iew, or the right bar, only the. sorting algorithm employed.

**Comment by javiergonzper** [comment]
The point is that if we change the algorithm to sort the files and folders we also have to modify the right bar because it will not have sense and will not work as we developed it if we put the numbers on the beginning.
The natural (and native) behaviour of a list of things on iOS is putting the numbers and the special characters at the end as you can see on the "Contacts".

You have to take into account that this is iOS not a Windows or MacOS file system.

**Comment by zinks-** [comment]
iOS has inconsistent file sorting behavior: "App store" results show numbers first, "Videos" show numbers last,.. So there is no "iOS" way to do it, really!

**Comment by ponychicken** [comment]
@zinks- I think it is simply the realisation that different types of data have different properties. App names often contain numbers, person names not so often etc.
